### PR TITLE
[p]slowmode

### PR DESCRIPTION
**Pull request by DevilXD**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes

This PR adds a `[p]slowmode [interval=0]` command, for easy slowmode management on a channel. This lets anyone with `Manage Channels` permission or a `Mod` role to change the slowmode setting.

Could be changed to require `0` to disable instead of having it there by default, this is up for discussion.

### CODE REVIEW

File: redbot/cogs/mod/slowmode.py
Comment (skeith) [inline redbot/cogs/mod/slowmode.py]: can you add human time format converter?

seconds, minutes, hours etc

Comment (DevilXD) [reply redbot/cogs/mod/slowmode.py]: As the input parameter? Or for the `Slowmode interval is now {interval} seconds.` feedback message?

Comment (skeith) [reply redbot/cogs/mod/slowmode.py]: both

**Comment by Jackenmen** [comment]
I don't believe it's for us to decide, if mods should be able to use this and I think this should be changed so only people with admin role or manage channels permission can use it and just leave it to Red users to decide if they want to allow it to mods role (which they can do by using permissions cog)

**Comment by DevilXD** [comment]
This last commit comes with a simple time converter per request, that will convert input like `5`, `5s`, `5m` or `5h` to appropriate time interval. `humanized_timedelta` from `utils` is now used to pretty-print the slowmode interval set. Addressing jack's concerns, I've also bumped the decorator to `admin_or_permissions`, with the intention that it could be overwritten with permissions later if needed as suggested.

A proper, fully fledged time interval converter could be used here, but I believe this is beyond this simple PR. Such converter could be really useful to everyone using Red in general, not only here in the Mod cog, and could be incorporated here at a later date when available.

File: redbot/cogs/mod/slowmode.py
Comment (DevilXD) [reply redbot/cogs/mod/slowmode.py]: Should be resolved with the recent commit.

**Comment by mikeshardmind** [comment]
Assuming this works properly (It looks solid, but I won't have a chance to test until later) I'm fine with merging prior to having a generalized timedelta converter in core and adding it and switching this to it later.

For anyone who goes to add that, the converter should likely be generalized in such a way in which valid units can be passed as well as an upper and lower bound on the time so that all of that is handled in the conversion step.

**Review by mikeshardmind — CHANGES_REQUESTED** [review]
This works great for an initial addition. I do have a small change request here:

image

If this is going to use a simplistic handling prior to there being a generalized converter, this should throw an error to not cause confusion, rather than allow it to succeed with an ambiguous meaning.

a tweak to the regex used, or using `re.fullmatch` should be reasonably sufficient for this on the short term until the generalized converter is done.

**Comment by DevilXD** [comment]
Right, I was using `match` instead of `search` already,  but forgot that it still requires an explicit line end. Should be as easy as adding `$` at the end of the regex pattern.

File: redbot/cogs/mod/slowmode.py
Comment (Tobotimus) [inline redbot/cogs/mod/slowmode.py]: Please use `_("message").format()` rather than f-strings for user-facing strings. f-strings fundamentally can't work with gettext due to order of evaluation.
```suggestion
            await ctx.send(_("Slowmode interval is now {interval}.").format(interval=interval))
```

File: redbot/cogs/mod/slowmode.py
Comment (mikeshardmind) [inline redbot/cogs/mod/slowmode.py]: These negative conditions should be handled separately. 

To a user, getting this feedback as a result of this input isn't clear what they did wrong. Sending the help on a non match is fine here.

image

**Review by mikeshardmind — DISMISSED** [review]

File: redbot/cogs/mod/slowmode.py
Comment (DevilXD) [reply redbot/cogs/mod/slowmode.py]: Right, switched it to the format method, slightly shorter than your version - hopefully that's alright.

File: redbot/cogs/mod/slowmode.py
Comment (DevilXD) [reply redbot/cogs/mod/slowmode.py]: This should be resolved once #2736 will be accepted, so I can use it here.

**Comment by DevilXD** [comment]
This should be ready to go now.

**Review by Tobotimus — APPROVED** [review]
Thanks for your contribution! Works well :)

**Pull request merged**
